config/routes.rb

```diff
@@ -303,15 +304,26 @@
       put '/cac/:step' => 'cac#update'
     end
     if Figaro.env.enable_mobile_capture == 'true'
-      get '/mobile_capture' => 'mobile_capture#new'
-      get '/AssureIDService/subscriptions' => 'assure_id_service#subscriptions'
-      post '/AssureIDService/Document/Instance' => 'assure_id_service#instance'
-      post '/AssureIDService/Document/:guid/Image' => 'assure_id_service#image'
-      get '/AssureIDService/Document/:guid/Classification' => 'assure_id_service#classification'
-      get '/AssureIDService/Document/:guid' => 'assure_id_service#document'
-      get '/AssureIDService/Document/:guid/Field/Image' => 'assure_id_service#field_image'
-      post '/api/v1/liveness' => 'assure_id_service#liveness'
-      post '/api/v1/facematch' => 'assure_id_service#facematch'
+      get '/verify/doc_auth_v2' => 'idv/doc_auth_v2#index'
+      get '/verify/doc_auth_v2/scan_id' => 'idv/scan_id#new'
+      get '/verify/doc_auth_v2/:step' => 'idv/doc_auth_v2#show', as: :idv_doc_auth_v2_step
+      put '/verify/doc_auth_v2/:step' => 'idv/doc_auth_v2#update'
+      get '/verify/doc_auth_v2/link_sent/poll' => 'idv/doc_auth_v2#doc_capture_poll'
+
+      get '/scan_id' => 'idv/scan_id#new'
+      get '/scan_complete' => 'idv/scan_id#scan_complete'
+      get '/capture/photo' => 'idv/scan_id#new'
+      get '/capture/camera' => 'idv/scan_id#new'
+      get '/photo/confirm' => 'idv/scan_id#new'
+      get '/capture/selfie' => 'idv/scan_id#new'
+      get '/AssureIDService/subscriptions' => 'idv/scan_id#subscriptions'
+      post '/AssureIDService/Document/Instance' => 'idv/scan_id#instance'
+      post '/AssureIDService/Document/:instance_id/Image' => 'idv/scan_id#image'
+      get '/AssureIDService/Document/:instance_id/Classification' => 'idv/scan_id#classification'
+      get '/AssureIDService/Document/:instance_id' => 'idv/scan_id#document'
+      get '/AssureIDService/Document/:instance_id/Field/Image' => 'idv/scan_id#field_image'
```

Comment:
These URLs seem like good candidates to move to a separate controller?
- they all have a URL param for an instance
- they all have nonstandard actions (ex not new, update, post, get, etc)

The controller is super big and a lot to read at once

Reply:
250 lines is super big?

Reply:
Yes, especially for our codebase. It's a 275 line file which puts it in the 99% percentile compared to the rest of our codebase in terms of size

```
# get controller sizes
find app/controllers -type f | grep -v concerns | xargs wc -l
```
```
# in R
> summary(x)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    8.0    19.0    46.0    55.7    73.0   295.0 
> quantile(x, c(.75, .90, .95, .98, .99))
   75%    90%    95%    98%    99% 
 73.00 115.40 140.40 182.16 226.24 
```

Reply:
Historically some of our fat controllers were slimmed down by taking out all the logic en masse and placed into a service or form (ugh) that would only ever be used in the context of that single  controller.  Not only did it increase the lines of code to be managed but it made the controllers harder to follow with all the indirection and a where's waldo to figure out what was actually happening.  If you can slim it down into reusable services I'm all for it...but to simply move the logic elsewhere because fat controllers are bad is dogma.

Reply:
I will try to extract out some services....

Reply:
Totally agree about dogma, but I think the fact that half the URLs have a param and "/Document` in them but half of them don't indicates there are some concerns we can easily separate

Reply:
In reality those are all the urls the Acuant sample code is using so I was trying to keep them isolated to one controller.

Reply:
The names are bad too.  I could change them but I'd have to hack the js.

Reply:
After looking at this I'm definitely liking the two controller approach to segment our stuff from Acuant's stuff.  Good call on that... it should break cleanly

Reply:
wow that was painful